packages/datadog-instrumentations/src/child_process.js

```diff
@@ -112,9 +199,39 @@ function wrapChildProcessAsyncMethod (shell = false) {
 
       const innerResource = new AsyncResource('bound-anonymous-fn')
       return innerResource.runInAsyncScope(() => {
-        childProcessChannel.start.publish({ command: childProcessInfo.command, shell: childProcessInfo.shell })
+        const abortController = new AbortController()
+        const { command, file, shell, fileArgs } = childProcessInfo
+        const context = {
+          command,
+          file,
+          shell
+        }
+        if (fileArgs) {
+          context.fileArgs = fileArgs
+        }
+
+        childProcessChannel.start.publish({ ...context, abortController })
+
+        let childProcess
+        if (abortController.signal.aborted) {
+          childProcess = new ChildProcess()
```

Comment:
why do you need to create a new `ChildProcess` and emit `error` and `close` events? 
couldn't you just call `cb(error)`?

Reply:
These methods are expecting to return a ChildProcess, they fails asynchronously. We can not return an object with less properties or methods than the original, and we can't throw an Error when the expected behaviour is calling the callback.
With childProcess, customer would subscribe to `childProcess.on('error', ...)` or `childProcess.on('close')` without callback, we should support all the use cases.

Reply:
ok, I hadn't seen the `return childProcess` :S